util/cass.py

```diff
@@ -576,6 +578,308 @@ def insert_dataset(stream_key, dataset):
     return ret_val
 
 
+@log_timing(log)
+def insert_dataset(stream_key, dataset):
```

Comment:
It would help readability if this function could be split into multiple functions.